### ISSUE TITLE
grf

### ISSUE THREAD

**Issue by aksenav2** [DS-Homebrew/twlmenu-extras#160]
Hello, please tell me how to open or edit images that are in skins for dsi in .grf format?

**Comment by aksenav2** [comment]
> Use the example skins linked on the [wiki page](https://wiki.ds-homebrew.com/twilightmenu/custom-dsi-3ds-skins.html)

Thanks for the advice, that's what I did, but not all skins suit me, some of the graphics I translated into my native language, and often in this format and no programs and editors can open it.

**Comment by spellboundtriangle** [comment]
The example skins don't use the GRF format, though

**Comment by aksenav2** [comment]
> Grab mine Pacman skin, or Yoshi skin by bony (which my skin is based on), I don't use GRF format at all, just plain BMP and PNG files.

Thanks for the tips and advice. Pac-man skin is simply gorgeous and looks cool.

**Comment by Allinxter** [comment]
> > 
> 
> I know, but then you need to draw completely from the very beginning, and I just wanted to edit an already finished skin. In general, as I understand it, you also do not know how to edit files in the format .grf ?

It's not really possible to do so, not easily at least. Think of grf as an already locked format. Images were converted from png and bmp to grf in order for the console to read them quickly, without needing to convert them first, and this conversion can't be reversed.

**Comment by 0x0ryba** [comment]
> > > 
> > 
> > 
> > I know, but then you need to draw completely from the very beginning, and I just wanted to edit an already finished skin. In general, as I understand it, you also do not know how to edit files in the format .grf ?
> 
> It's not really possible to do so, not easily at least. Think of grf as an already locked format. Images were converted from png and bmp to grf in order for the console to read them quickly, without needing to convert them first, and this conversion can't be reversed.

OK. So that's why loading time of my skin is longer...

**Comment by aksenav2** [comment]
Tell me one more question, how did you work with the font .NFTR, through a program or an online editor?

**Comment by 0x0ryba** [comment]
Yes. As I didn't figure out, how to copy characters across different nftr fonts, I draw all characters from scratch (well, not all characters included in font, only latin aphabet letters and numbers and some ,.-/ characters) 4 times. As DSi fonts renamed to DS crashed me on DS.
PK11 also took me some advices in https://gbatemp.net/threads/ds-i-3ds-twilight-menu-gui-for-ds-i-games-and-ds-i-menu-replacement.472200/page-772
so,check it out.
But making custom font was quite long process...

**Comment by 0x0ryba** [comment]
Oh,sorry, I didn't answer your question...
I used PK11's online editor ;)
